### DISCUSSION TITLE
Popen?

### DISCUSSION THREAD

**Discussion by misha-franz** [red/red#5550] [Q&A] [answered]
Is there a way in Red to open an external app (for example ssh) as a channel/pipe and communicate with it via standard input/output?

**Comment by hiiamboris** [comment, marked as answer]
As Red has no official async I/O to date, you can only lock your program until the external app finishes:
```
>> call/shell/output "echo %CD%" out: {}
== 0                                            ;) 0 indicates success
>> ? out
OUT is a string! value: "d:\devel\red^/"        ;) stdout captured
```

**Reply by misha-franz** [reply]
Thanks for your quick reply.  Is there unofficial async IO?  Also maybe there are plans to support ssh directly (?)

**Reply by hiiamboris** [reply]
Unofficial WIP is [here](https://github.com/red/red/tree/IO/), but I can't speak about its file and especially *standard* I/O support. Seems littered with debug probes 🤷‍♀️
I doubt ssh will be supported in the core, but it's possible that someone will write it as a module eventually as `ssh://` port.
